src/main/java/org/opensearch/neuralsearch/query/HybridQuery.java

```diff
@@ -46,7 +46,8 @@ public HybridQuery(final Collection<Query> subQueries, final List<Query> filterQ
         if (subQueries.isEmpty()) {
             throw new IllegalArgumentException("collection of queries must not be empty");
         }
-        if (hybridQueryContext.getPaginationDepth() == 0) {
+        Integer paginationDepth = hybridQueryContext.getPaginationDepth();
+        if (paginationDepth != null && paginationDepth == 0) {
             throw new IllegalArgumentException("pagination_depth must not be zero");
```

Comment:
Do you want to log the actual value to understand if it's null or 0?
does this error message hold if the value is null? If not can you extend it and include the `null` scenario so user can understand and fix the problem.

Reply:
The value will never be null to the versions greater that or equal to 2.19.  In case if user didn't send pagination_depth in os >=2.19 then it will take the default value 10 [code ref](https://github.com/opensearch-project/neural-search/pull/1094/files#diff-e777f6753c5496193dda2794e4be74162f5d2cbb5c9f278df5cbb4bf5438a6e2R169).

Reply:
understood about the scenario. Please add the log the message part, it can be at the error level as we will fail with runtime exception in that case

Reply:
I will address this in https://github.com/opensearch-project/neural-search/issues/1097

Reply:
Synced up offline and concluded that we do not need it